### PR TITLE
[android] update to the latest AndroidX packages

### PR DESCRIPTION
**Pull request by jonathanpeppers**
Context: https://github.com/dotnet/maui/pull/5305
Fixes: https://github.com/xamarin/AndroidX/issues/509

Update to the latest AndroidX packages, so we also use the latest
Xamarin.Google.Guava.ListenableFuture

This currently has build failures:

```
src\Core\src\Platform\Android\Navigation\StackNavigationManager.cs(145,41): error CS1061: 'NavController' does not contain a definition for 'BackStack' and no accessible extension method 'BackStack' accepting a first argument of type 'NavController' could be found (are you missing a using directive or an assembly reference?)
src\Core\src\Platform\Android\Navigation\StackNavigationManager.cs(188,48): error CS1061: 'NavController' does not contain a definition for 'BackStack' and no accessible extension method 'BackStack' accepting a first argument of type 'NavController' could be found (are you missing a using directive or an assembly reference?)
src\Core\src\Platform\Android\Navigation\StackNavigationManager.cs(245,51): error CS1061: 'NavController' does not contain a definition for 'BackStack' and no accessible extension method 'BackStack' accepting a first argument of type 'NavController' could be found (are you missing a using directive or an assembly reference?)
```

Looking at:

    ~\.nuget\packages\xamarin.androidx.navigation.runtime\2.4.1\lib\net6.0-android31.0\Xamarin.AndroidX.Navigation.Runtime.dll

I can't find:

    [Register("androidx/navigation/NavController", DoNotGenerateAcw = true)]
    public class NavController : Object
    {
        public unsafe virtual IDeque BackStack
        {
            [Register("getBackStack", "()Ljava/util/Deque;", "GetGetBackStackHandler")]
            get;

But it's not documented here either:

https://developer.android.com/reference/androidx/navigation/NavController

### CODE REVIEW

**Comment by jpobst** [comment]
It does appear that it has been removed in 2.4+.

navigation-runtime-2.3.5.aar:
image

navigation-runtime-2.4.1.aar:
image

I think their:
```
@RestrictTo({RestrictTo.Scope.LIBRARY_GROUP})
```

essentially means "this is not public API".

[Documentation](https://developer.android.com/reference/androidx/annotation/RestrictTo.Scope)
```
LIBRARY_GROUP - Restrict usage to code within the same group of libraries.
```

We likely should not be binding stuff marked this way, but it's not something our tools have support for.

**Comment by jonathanpeppers** [comment]
@jpobst should MAUI just be using a different API here?

@PureWeen it looks like maybe you wrote this part, do you see what we should change?

**Review by moljac — APPROVED** [review]
version bumps - LGTM

File: src/Core/src/Platform/Android/Navigation/StackNavigationManager.cs [hunk 1]
```diff
@@ -142,7 +143,7 @@ void ApplyNavigationRequest(NavigationRequest args)
 
 			IsAnimated = animated;
 
-			var iterator = NavHost.NavController.BackStack.Iterator();
+			var iterator = NavHost.NavController.BackQueue.Iterator();
```
Comment (jonathanpeppers) [inline src/Core/src/Platform/Android/Navigation/StackNavigationManager.cs:146]: LOL!

**Comment by rmarinho** [comment]
```
   (_ReadAndroidManifest target) -> 
         obj\Release\net6.0-android\android\AndroidManifest.xml : error XA4218: Unable to find //manifest/application/uses-library at path: C:\Program Files (x86)\Android\android-sdk\platforms\android-31\optional\androidx.window.extensions.jar [D:\a\_work\_temp\templatesTestmaui\templatesTestmaui.csproj]
         obj\Release\net6.0-android\android\AndroidManifest.xml : error XA4218: Unable to find //manifest/application/uses-library at path: C:\Program Files (x86)\Android\android-sdk\platforms\android-31\optional\androidx.window.sidecar.jar [D:\a\_work\_temp\templatesTestmaui\templatesTestmaui.csproj]
```

**Comment by jonathanpeppers** [comment]
I'm having trouble reproducing this problem locally... I get no warnings or errors with `../bin/dotnet/dotnet new maui`.

Trying `$(MSBuildWarningsAsMessages)` now.

**Comment by jonathanpeppers** [comment]
Ok, I see the same test failures on other PRs. I had retried them a few times.

**Pull request merged**